apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/faucet/components/SpinWheel.tsx

```diff
@@ -70,41 +69,51 @@ export function TestnetSpinWheel({
     chain: definedChain,
     client: thirdwebClient,
   });
-  const [fundingSent, setFundingSent] = useState(false);
   const router = useRouter();
 
-  const requestFunds = async (result: string) => {
-    try {
-      const response = await fetch("/api/testnet-faucet/claim", {
-        method: "POST",
-        body: JSON.stringify({
+  const claimMutation = useMutation({
+    mutationFn: async (result: string) => {
+      try {
+        const amount = result === "Come back tomorrow" ? "0" : result;
+        const requestBody = {
           chainId: chainId,
           toAddress: address,
-          amount: result === "Come back tomorrow" ? "0" : result,
-        }),
-      });
-
-      if (response.ok) {
-        toast.success(
-          `${result} ${chain?.nativeCurrency?.name} sent to ${address}`,
-        );
-        setFundingSent(true);
+          amount: amount,
+        };
+
+        const response = await fetch("/api/testnet-faucet/claim", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(requestBody),
+        });
+
+        if (!response.ok) {
+          const errorText = await response.json();
+          toast.error(errorText.error);
+          return;
+        }
+
         router.refresh();
-      } else {
-        const errorText = await response.json();
-        toast.error(errorText.error);
+        const successMessage = `${amount} ${chain?.nativeCurrency?.name} sent to ${address}`;
+        toast.success(successMessage);
+      } catch (error) {
+        toast.error("An unexpected error occurred.");
```

Comment:
`claimMutation.isSuccess` will always be `true` if you don't re-throw error here